### DISCUSSION TITLE
Give reasons on why the type is not correct

### DISCUSSION THREAD

**Discussion by edobrb** [samchon/typia#118] [Ideas]
Hello,
would it be possible to add an explanation on why the `.is` or the `.assertType` failed?
Something like [io-ts](https://github.com/gcanti/io-ts/blob/master/index.md#custom-error-messages) which in the event of an error has a data structure describing why the type checking failed.

For example, in a call to service I expect to receive a JSON of type T, in the case `.is` returns false i would like to give to the end user an explanatory error.

**Comment by samchon** [comment]
`TypedRouteError` thrown by `TSON.assertType()` is providing `path` and `value` properties.

What you want is getting the reason by returned value instead of the `TypedRouteError`? Not enough it is?

![118](https://user-images.githubusercontent.com/13158709/175948651-47d2483f-4a31-4645-a5f3-ab93a5f63066.png)

**Comment by samchon** [comment]
If make a new function, its name would be `validate` and would return array of reasons.

Can you make an issue and design the reason structure as an object structure?

Implementation would not hard and may be possible at this Saturday.

**Reply by edobrb** [reply]
Thanks for the quick reply, ASAP i'll try to design the error structure and i open a new issue.

**Comment by samchon** [comment]
Have implemented
